lib/child_process.js

```diff
@@ -266,7 +269,8 @@ function execFile(file /* , args, options, callback */) {
     uid: options.uid,
     shell: options.shell,
     windowsHide: !!options.windowsHide,
-    windowsVerbatimArguments: !!options.windowsVerbatimArguments
+    windowsVerbatimArguments: !!options.windowsVerbatimArguments,
+    signal: options.signal
```

Comment:
I think if you want spawn to work here you'd need to use spawnWithSignal and not spawn, but I don't think that'd work since `kill` below also deals with exec specifics.

Reply:
Yeah, see the test failures below

Reply:
Sure, basically it's getting called twice. I made a change on 8678797
Not totally sure if it's the right way to go, but fixes the double emitting of the event, please let me know your thoughts